aya/src/programs/perf_attach.rs

```diff
@@ -72,10 +71,10 @@ impl Link for PerfLink {
     }
 }
 
-pub(crate) fn perf_attach(prog_fd: RawFd, fd: RawFd) -> Result<PerfLinkInner, ProgramError> {
+pub(crate) fn perf_attach(prog_fd: RawFd, fd: OwnedFd) -> Result<PerfLinkInner, ProgramError> {
     if FEATURES.bpf_perf_link() {
         let link_fd =
-            bpf_link_create(prog_fd, fd, BPF_PERF_EVENT, None, 0).map_err(|(_, io_error)| {
+            bpf_link_create(prog_fd, fd.as_raw_fd(), BPF_PERF_EVENT, None, 0).map_err(|(_, io_error)| {
```

Comment:
Prior to this PR this would leak the file descriptor as we would drop it at the end of this scope without closing it and no one else had a handle to it. Now at the end of this scope the `drop` will close the file descriptor for us.